src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface DefaultLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface SidebarLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The width of the area for the sidebar. */
+    sidebarWidth?: string | number;
+    /** If the sidebar is on the left or the right of the layout */
+    sidebarAlign?: SidebarAlignment;
+    /** The actual sidebar. This can be a component. */
+    sidebar?: any;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
     className?: string;
     children?: any;
 }
 
-export const Page = ({ children }: PageProps): React.ReactElement<any> => (
-    <StyledPageElement className="anchor-page">
+interface StyledLayoutProps {
+    layoutWidth?: string | number;
+    layoutBackgroundColor?: string;
+}
+
+interface StyledContentProps {
+    contentWidth?: string | number;
+    contentBackgroundColor?: string;
+}
+
+// Components
+// ------------------------------------------------------------------------------------------------------------------
+export const Page = (props: PageProps): React.ReactElement<any> => (
+    <StyledPage className={classnames('anchor-page', props.className)}>
         <NormalizeCSS />
         <GlobalCSS />
-        <Header />
-        <StyledBodyElement>{children}</StyledBodyElement>
-        <Footer />
-    </StyledPageElement>
+        <Grid
+            columns={1}
+            rows={`minmax(${props.headerHeight},auto) 1fr minmax(${
+                props.footerHeight
+            },auto)`}
+            height={'100vh'}
+        >
+            {props.header && <Cell>{props.header}</Cell>}
+
+            <Cell>{props.children}</Cell>
+
+            {props.footer && <Cell>{props.footer}</Cell>}
+        </Grid>
+    </StyledPage>
+);
+
+Page.defaultProps = PageDefaultProps;
+
+const DefaultLayout = (
+    props: DefaultLayoutProps
+): React.ReactElement<any> => (
+    <StyledLayout
+        layoutWidth={props.layoutWidth}
+        layoutBackgroundColor={props.layoutBackgroundColor}
+        className={classnames('anchor-default-layout', props.className)}
```

Comment:
Does this need to be `default-layout` or can it just be layout? It's our only layout so default is kind of implied IMO.

Reply:
There's two layouts, DefaultLayout and SidebarLayout. I went this route because this naming gives us latitude if we wanted to add more layouts. I think the door should be left open for that possibility, because although it's all we have right now that doesn't mean the list won't get bigger.